app/controllers/publications/overview/shortlist.js

```diff
@@ -0,0 +1,127 @@
+import CONSTANTS from 'frontend-kaleidos/config/constants';
+
+export default class PublicationsOverviewShortlistController extends Controller {
+  @service publicationService;
+  @service router;
+  @service store;
+
+  @tracked isOpenNewPublicationModal;
+  pieceForPublication;
+  agendaitemForPublication;
+  decisionActivityForPublication;
+  caseForPublication;
+  mandateesForPublication;
+
+  getAgendaitem = async (piece) => {
+    const agendaitems = await piece.agendaitems;
+    let agendaitem;
+    for (let maybeAgendaitem of agendaitems) {
+      const agenda = await maybeAgendaitem.agenda;
+      const nextVersion = await agenda.nextVersion;
+      if (!nextVersion) {
+        agendaitem = maybeAgendaitem;
+        break;
+      }
+    }
+    return agendaitem;
+  }
+
+  getDecisionDate = async (piece) => {
+    const agendaitem = await this.getAgendaitem(piece);
+    const treatment = await agendaitem.treatment;
+    const decisionActivity = await treatment.decisionActivity;
+    const decisionDate = await decisionActivity.startDate;
+    return decisionDate;
+  }
+
+  getMandateeNames = async (piece) => {
+    const agendaitem = await this.getAgendaitem(piece);
+    const mandatees = await agendaitem.mandatees;
+    const persons = await Promise.all(
+      mandatees.map((mandatee) => mandatee.person)
+    );
+    return persons.map((person) => person.fullName);
+  }
+
+  @action
+  async openNewPublicationModal(piece) {
+    this.pieceForPublication = piece;
+    this.agendaitemForPublication = await this.getAgendaitem(piece);
+    this.mandateesForPublication = await this.agendaitemForPublication.mandatees;
+
+    const treatment = await this.agendaitemForPublication.treatment;
+    this.decisionActivityForPublication = await treatment.decisionActivity;
+
+    const subcase = await this.decisionActivityForPublication.subcase;
+    const decisionmakingFlow = await subcase.decisionmakingFlow;
+    this.caseForPublication = await decisionmakingFlow.case;
+
+    this.isOpenNewPublicationModal = true;
+  }
+
+  @action
+  closeNewPublicationModal() {
+    this.pieceForPublication = null;
+    this.agendaitemForPublication = null;
+    this.decisionActivityForPublication = null;
+    this.mandateesForPublication = null;
+    this.caseForPublication = null;
+
+    this.isOpenNewPublicationModal = false;
+  }
+
+  @action
+  async saveNewPublication(publicationProperties) {
+    const regulationType = await this.getRegulationType(this.pieceForPublication);
+    const publicationFlow =
+      await this.publicationService.createNewPublicationFromMinisterialCouncil(
+        publicationProperties,
+        {
+          case: this.caseForPublication,
+          decisionActivity: this.decisionActivityForPublication,
+          mandatees: this.mandateesForPublication,
+          regulationType: regulationType,
+        }
+      );
+    this.pieceForPublication.publicationFlow = publicationFlow;
+    await this.pieceForPublication.save();
```

Comment:
Saving the piece here.
We do this for publications I see (this is also in older code).
This could lead to concurrency issues.
Perhaps user 1 is starting a pub-flow while user 2 is updating the access right to public.
Also, with the default save override we also change `modified` and `accessLevelLastModified` properties by doing this.
Could we just create `pub-flow` with a list (of 1) instead so we set the relation on the inverse?